### PR TITLE
Fix CI builds on internal branches

### LINKED ISSUE #602: CI passes-through fails of builds in inside repo branches (`master`)
During CI development GHC 8.10 never successfully passed.

On `master` it passes. on PRs out of the tree - do not passes. So it is probably related to the `| cachix push` part that I've could not tested out of the tree.

But the `| cachix push` was already in `master`, I didn't changed its process.

Also what is reporting "All done" on failure :thinking:

I would look tomorrow (today).

It is or shell pipe related or something in the script, or some upstream bug.

There is pretty tight strict conditions in a `shell` script:
```
set -xe
set -euo pipefail
```

### PR DESCRIPTION
**Pull request by Anton-Latukha**
Report #602

### CODE REVIEW

**Comment by Anton-Latukha** [comment]
Build https://travis-ci.org/github/haskell-nix/hnix/jobs/692903048 is proof that this fix already works.

It is probably due to `set` was in function scope, I already forgot are `shell` functions are functions or export & fork processes. This shows that it is most probably the second, and was expecting the first.

**Review by sjakobi — DISMISSED** [review]
Thanks for working on this! I'll approve already so you can merge once you're ready.

Please shout if you need another approval.

**Comment by Anton-Latukha** [comment]
As it is WIP, there would be rebasing.

As the main code landed and now I created an internal branch and build have proper caches - I can tweak and check several things pretty quickly that everything is working properly.

In epic PR that would've been just to much of everything at the same time.

**Comment by Anton-Latukha** [comment]
@sjakobi 

Also, you left a review with comments and not approval. Approval needs to be set manually, and it is the central option _afair_.
That is why I observed that you approved the PR verbally, but in fact, accidentally left a review requiring changes. As you also already said previously.

**Review by sjakobi — DISMISSED** [review]
@Anton-Latukha I'm pretty sure I approved this earlier, but your later pushes dismissed it:

image

**Comment by Anton-Latukha** [comment]
Oh, yep.
I've read your message too late and so when I looked at the Reviewers field I said what I've seen there.

**Comment by Anton-Latukha** [comment]
Before - I not had an opportunity to benchmark Cachix cache vs Travis, since out of the tree I not had the opportunity to create and heat-up a Cachix cache.

Now on battle test in the same conditions: 

With heated-up Travis cache: https://travis-ci.org/github/haskell-nix/hnix/builds/692906403 - 3 main builds time sum `~40.00 min`
Completely without Travis cache: https://travis-ci.org/github/haskell-nix/hnix/builds/692909205 - 3 main builds time sum `~22.30 min`

Now it is obvious that configuration without Travis cache is way better, in fact, two times faster.

**Comment by Anton-Latukha** [comment]
It is sad that such a good company, made so much for the Open Source movement, and just could not progress over the years due to holding their good promises.

**Comment by Anton-Latukha** [comment]
Insane.

Seems like there is no simple way to provide attributes through `--attr` option. Key just does the other thing.

The only way there is seems to be:

```
nix-build -E 'with (import <nixpkgs>{}); nginx.override { openssl = libressl; }'`
```

Which is insane talking. I need to compose string like this in CI.

It seems easier to just tweak up requirements in `default.nix` directly.

After all, we have `default.nix` as a basic developer ground - it seems logical to keep requirements closer to development. But I would wonder, would Nix loaded package in some `ghcid` or `hie-lsp` fall on turned-on `failOnAllWarnings` requirement for example. 

Honestly, I can just debug and compose this string inside the `shell` script moderately easy, but this is insane in itself.

To the build tool, the configs attributes can not be provided through `--attr` not through `--arg`, insane.

I kind of remember such thing about `nix-build`, I kind of relearn it from year to year. But man, every time I return I feel like - this should be, there just should be the way - and there is just not.

We would need to expose all keys of `haskell.lib` in `default.nix` explicitly.

**Comment by Anton-Latukha** [comment]
I would expose the API in different PR, this one would fix `master` builds.

**Comment by Anton-Latukha** [comment]
This PR is ready. The proof: on internal repo branch GHC 8.10 status is right: https://travis-ci.org/github/haskell-nix/hnix/builds/693021087
```
Setup: Encountered missing or private dependencies:

base >=4.9 && <4.14, template-haskell >=2.11 && <2.16
...
All done.

The command "if [ ! "$GHCVERSION" = 'ghcjs' ]; then ./build.sh; fi" exited with 100.
```

File: .travis.yml
Comment (sjakobi) [inline .travis.yml:158]: Why remove the caching? Can you say roughly how much this slows down CI?

File: build.sh
Comment (sjakobi) [inline build.sh:32]:
```suggestion
ghcjsLogTailLength=${ghcjsLogTailLength:-'10000'}

```

**Review by sjakobi — DISMISSED** [review]
I have no idea how this works, but if it makes CI results more reliable, that's great! :)

File: .travis.yml
Comment (Anton-Latukha) [reply .travis.yml:158]: Already discussed this: https://github.com/haskell-nix/hnix/pull/604#issuecomment-636348096, also commit name might be a hint :eyes:.

**Comment by Anton-Latukha** [comment]
> I have no idea how this works, but if it makes CI results more reliable, that's great! :)

It is basically understanding `set -Eexuo pipefail` in `shell`.

There some work would be ahead in other PRs to make all those CI options work, they did not worked as I expected, and thise Nix options do not work seemlessly, it needs to be adjested further.

**Review by sjakobi — APPROVED** [review]

**Pull request merged**
